src/Mealz/MealBundle/Service/ParticipationServiceTrait.php

```diff
@@ -98,26 +100,68 @@ private function slotIsAvailable(Slot $slot, DateTime $date): bool
         return $slotPartCount < $slotLimit;
     }
 
+    /**
+     * @throws ParticipationException
+     */
     private function createParticipation(Profile $profile, Meal $meal, ?Slot $slot = null, array $dishSlugs = []): Participant
     {
-        $participation = new Participant($profile, $meal);
+        $participant = new Participant($profile, $meal);
         if (null !== $slot) {
-            $participation->setSlot($slot);
+            $participant->setSlot($slot);
+        }
+
+        if ($meal->isCombinedMeal()) {
+            $this->updateParticipation($participant, $dishSlugs);
+        }
+
+        return $participant;
+    }
+
+    /**
+     * @throws ParticipationException
+     */
+    private function updateParticipation(Participant $participant, array $dishSlugs): void
```

Comment:
Same here. The name doesn't describe what the method does.

Reply:
Updated here too.